app/uk/gov/hmrc/nisp/views/includes/helpers/error_overview.scala.html

```diff
@@ -0,0 +1,16 @@
+@(form: Form[_])
+
+@if(form.hasErrors) {
+        <div id="error-summary-display" class="flash error-summary error-summary--show " role="group" aria-labelledby="error-summary-heading" tabindex="-1">
+            <h2 id="error-summary-heading" class="h3-heading">@Messages("nisp.validation.summary.heading")</h2>
+            <ul class="js-error-summary-messages">
+                @form.errors.map { error =>
+                <li role="tooltip">
+                    <a href='#@error.key' id='@error.key-error-summary' data-focuses='@error.key'>
+                        @if(error.key=="email") {@Messages("nisp.validation.summary.email") }else{ @Messages(error.message, error.args : _*)}
```

Comment:
Ok so I've done some digging.
### In regards to email code
To override the email validation message, you can simply add an entry to the messages file called "error.email".  This removes the needs for the if statement in `error_overview.html`

### In regards to the maximum limit error summary
You can do a similar thing and override `error.maxLength` but obviously this will hardcore all of the forms in the service to say this.
We have some options:
1. Continue overriding play-ui's error summary with our own allowing it to pass in arguments. In this case I think we should rename our `error_overview` to be `errorSummary` to mimic play UI as closely as possible.
2. Keep the play-ui implementation and override the error in the QuestionnareController to have our custom message. (This opens up an avenue to have a more powerful message like @fred-jackson wanted). This would allow ` @if(form.errors("improve").nonEmpty) {<span class="error-notification">@Messages("nisp.validation.summary.error.max")</span>}` to become cleaner too.

I prefer Option 2. What does @prashantchoudhari think